operator/pkg/component/component/component.go

```diff
@@ -73,12 +73,10 @@ type IstioComponent interface {
 	RenderManifest() (string, error)
 }
 
-// CommonComponentFields is a struct common to all components.
-type CommonComponentFields struct {
+// Component is a struct common to all components.
+type Component struct {
```

Comment:
this name stutters. also in general i'm not sure it's a good idea to promote the struct name to be the first class citizen - usually in go the interface gets naming priority and if there's an associated struct it ends up being called something else e.g. ComponentFields/Struct/whatever.
if i understand correctly where you're going with this pr, i propose the following naming: 
 - create a new type called IstioCoreComponent which implements the IstioComponent interface (rename NewComponent to NewIstioCoreComponent below)
- call this struct IstioComponentStruct or something like that (it's a pity it can't be called IstioComponent like the interface but that's just a go quirk)

Reply:
Renamed to `IstioComponentImpl`, cpp-style